erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -167,10 +167,20 @@ def on_cancel(self):
 
 		self.check_nextdoc_docstatus()
 		self.update_reserved_qty()
-
+		self.update_project()
 		self.update_prevdoc_status('cancel')
 
 		frappe.db.set(self, 'status', 'Cancelled')
+		
+	def update_project(self):
+		project_list = []
+		for d in self.items:
```

Comment:
"Project" field does not exists in Sales Order Item table, it is in the parent table.